### DISCUSSION TITLE
Not seeing how to get MySQL 5 driver

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#16523] [Q&A]
I ran into some weird Java errors importing a MySQL 5.6 DB to an existing local one, but I've been using the 5.7 "Local Client", and since I couldn't figure out how to fix those errors, I just ended up deleting the "driver".

Now that I'm trying to re-set it up, but this time using MySQL 5.6 "Local Client", I'm in a maze trying to figure out how to just get this working again.

File > Properties just showed details for the currently-selected DB; not useful.

I saw nothing in DBeaver > Preferences to fix this.

When I clicked Database > Driver Manager, the only option for MySQL says "Driver for MySQL 8 and later"... but, I need the driver for MySQL 5.

![image](https://user-images.githubusercontent.com/74785256/169070893-6b632300-1a13-447e-9f95-6e7c141eb3a5.png)

EDIT: I'm not sure what the # column number refers to, it's a little too cryptic... driver version number?

I think I was using this driver previously when things were working fine (mostly for MySQL 5.6 databases, only one was MySQL 8, and I only used that randomly for a few months). I can't tell what started going wrong, nor how to just get this working for my MySQL 5.6 projects. 

PS: I'm sorry that all of my projects now are MySQL 5.6 😆 but they're paying the bills, and this job is the most convenient for me. Not what we're taught in school 😆 but this is the reality.

**Comment by givihuda** [comment]
Oh, now that I'm trying to figure out how to just _add the driver to make my databases work again_, there doesn't seem to be any way to do that from that screen I showed above. There's no "Save" or "Confirm" or "Use This" button; clicking Edit just shows me some cryptic settings that I don't know should be modified at all, but in a window that truncates the tabs for some reason...

![image](https://user-images.githubusercontent.com/74785256/169072419-89884230-fa6b-4c81-a25c-ea27f9e74bdc.png)

Like, what am I supposed to do now?

![image](https://user-images.githubusercontent.com/74785256/169072523-32aeefa1-41c8-4aec-b439-3cd84a671aab.png)
![image](https://user-images.githubusercontent.com/74785256/169072565-255388b1-2949-482e-ba2d-7a1f391db5a0.png)
![image](https://user-images.githubusercontent.com/74785256/169072588-484e029f-57e1-4e02-85ad-e0916550eb6e.png)
![image](https://user-images.githubusercontent.com/74785256/169072620-b801415f-2435-4758-900a-5fed5d2bed1d.png)

I think I may have deleted the Home folder, but that was /Applications/MAMP/Library/bin, and re-selecting that shows Product Version 5.7.34, which I don't want to use anymore... I suppose I need to just change MAMP's version to MySQL 5.6?

I tried adding it back:

![image](https://user-images.githubusercontent.com/74785256/169073500-0edb2b24-a768-4fd8-8e55-1d3e144947c2.png)

I think this is correct 🤔 However, these look like the same settings when I was having the import errors. The export was from a DB running on MySQL 5.6, and my DB for the import apparently runs on MySQL 5.7.

I'm sorry if I'm confused, maybe I don't know enough, but trying to figure out the problems and fix them has not been the most clear experience here.

**Reply by givihuda** [reply]
After doing what I thought was fixing it, restarting DBeaver now gives this error:

![image](https://user-images.githubusercontent.com/74785256/169075393-66c89c7a-74e7-4e73-8ff7-e36e8a48143c.png)

It suggests: "You should configure jars in driver settings."

OK... where? I thought I was already in there, but it was called "Driver Manager", and there's no "driver settings" menu option anywhere. What does it mean exactly?

**Comment by LonwoLonwo** [comment]
Did you try to connect with the tab "New connection"? We have separate MySQL 5 driver there.

![2022-05-19 09_10_09-Connect to a database](https://user-images.githubusercontent.com/45152336/169222763-fb83efb3-f377-4256-8a55-c1bb14b12c4e.png)

**Reply by givihuda** [reply]
How did you get that? I don't seem to have that there.

![image](https://user-images.githubusercontent.com/74785256/176235221-cecde049-fcc8-4df5-a4bf-470d200bb5e9.png)

**Reply by LonwoLonwo** [reply]
@givihuda Please choose the "All" tab in the left list to see all drivers

**Reply by givihuda** [reply]
Sorry, you're correct, I see that now.

![image](https://user-images.githubusercontent.com/74785256/202765122-3d4f315d-62a5-4b51-b8a7-54f3bd3dd2c3.png)

Anyway, now some new questions:

1. What does "MySQL (old)" refer to? I see no option to view what it means, but if I click 'Next' then the Driver Properties tab, it gives me this: 
![image](https://user-images.githubusercontent.com/74785256/202765458-ef105e65-e498-4b7c-98f8-2899263b8e52.png)
I'm clicking around there but I feel like I'm in a new rabbit hole and I'm not sure I'm understanding much about it. I suppose "(old)" refers to an older version of the driver, but the links I'm given don't seem to say much - https://www.mysql.com/products/connector/ - Do I even really need to care about this?
2. What's the difference between MySQL, MySQL (old), and MySQL 5?
3. Let me put it this way: I work with MySQL 5 or 8, sometimes in the same day... right now MySQL 5.7 seems to be the most compatible, but I'd prefer to have separate drivers where one could be set to work with the connection, and I will be looking into having MySQL 5.6 and MySQL 8.x installed at the same time (where I can flip between servers or have them run simultaneously). Is DBeaver able to work with this kind of setup?

**Reply by LonwoLonwo** [reply]
"MySQL (old)" - it is more about the code base. The old version depends on the driver and does all metadata reading, for example, throw drivers methods. 
Not old MySQL versions use code from this bundle: https://github.com/dbeaver/dbeaver/tree/devel/plugins/org.jkiss.dbeaver.ext.mysql

Drivers and database versions in the case of MySQL5 and MySQL8 are different. Partially. So MySQL database version 5 will work better with the fifth driver.

I use in my work different drivers for MySQL 5.5 and MySQL 8.0. I do not see a reason to use one driver for both.

**Comment by jsvachon2** [comment]
I have the same issue here with a fresh install of DBeaver. The old driver is missing from the list

**Reply by LonwoLonwo** [reply]
What kind of driver? From what list?

**Reply by jsvachon2** [reply]
Mysql 5 driver. Apparently, It does not ship with DBeaver anymore. I was able to add it to DBeaver on a Windows based machine but the same procedure does not seem to work on a Mac.

**Reply by LonwoLonwo** [reply]
![2022-11-22 19_24_21-Connect to a database](https://user-images.githubusercontent.com/45152336/203367575-19637c51-cb54-47b9-a779-d6edb529c797.png)

I can see MySQL 5 driver in the list in the last DBeaver release.

**Comment by JTrueUALR** [comment]
I just installed the latest update Version 23.2.5.202311191730 and do not see the MySQL 5 driver (on a Mac)

![image](https://github.com/dbeaver/dbeaver/assets/94559458/71522157-bdf4-44e3-8b40-810f77736036)

**Reply by LonwoLonwo** [reply]
Hello @JTrueUALR 

Please check our latest Early Acces version: https://dbeaver.io/files/ea/

**Reply by jltrue** [reply]
Thank you
